sale_timesheet_invoice_description/models/account_move.py

```diff
@@ -26,19 +28,19 @@ def _link_timesheets_to_invoice_line(self, start_date=None, end_date=None):
                     ("project_id", "!=", False),
                     ("timesheet_invoice_id", "=", aml.move_id.id),
                 ]
-                if start_date:
-                    domain = expression.AND([domain, [("date", ">=", start_date)]])
-                if end_date:
-                    domain = expression.AND([domain, [("date", "<=", end_date)]])
+                if date:
+                    domain = expression.AND([domain, [("date", ">=", date)]])
 
                 timesheets = self.env["account.analytic.line"].sudo().search(domain)
                 timesheets.write({"timesheet_invoice_line_id": aml.id})
 
-    def _check_balanced(self):
+    @contextmanager
```

Comment:
Why adding this? And why we need to override this check in first place? Maybe the reason has gone or it should be implemented in other way.

Reply:
It comes from this commit https://github.com/OCA/account-invoicing/commit/98c253e32b508207b31171a8bce6be96537cf7a1 . It looks like splitting the account move line by timesheet breaks the debit/credit balance by line.

Reply:
I'm afraid I don't know account.move well enough to suggest a better way.

Reply:
The new (in comparison to v15) `@contextmanager` here is needed, because the [super method in the Odoo v16 core](https://github.com/odoo/odoo/blob/3b24a13ee217b2d3846ad0953ccf8885d7b1f15a/addons/account/models/account_move.py#L1714-L1738) got one in the reworked accounting code, in commit https://github.com/odoo/odoo/commit/d8d47f9ff8554f4b39487fd2f13c153c7d6f958d.

However, the super call is implemented incorrectly here (a couple of lines below). It should read something line `with super()._check_balanced(container): yield`. The [flake8-bugbear warning B901](https://github.com/PyCQA/flake8-bugbear#opinionated-warnings), suppressed here with `# noqa`, points to that issue.

👉 I am fixing this.